team/bundles/org.eclipse.compare/compare/org/eclipse/compare/contentmergeviewer/TextMergeViewer.java

```diff
@@ -2059,14 +2059,7 @@ protected void createControls(Composite composite) {
 		// 1st row
 		if (fMarginWidth > 0) {
 			fAncestorCanvas = new Canvas(composite, SWT.DOUBLE_BUFFERED);
-			fAncestorCanvas.addPaintListener(new PaintListener() {
-
-				@Override
-				public void paintControl(PaintEvent e) {
-					paintSides(e.gc, fAncestor, fAncestorCanvas, false);
-
-				}
-			});
+			fAncestorCanvas.addPaintListener(e -> paintSides(e.gc, fAncestor, fAncestorCanvas, false));
```

Comment:
To keep the code comprehensible, it makes sense to give the variable a proper name when replacing the anonymous class with a lambda expression, as the explicit mentioning of the type is lost).

Reply:
👍 
See https://github.com/eclipse-platform/eclipse.platform/pull/1320#issuecomment-2061304178